### PR TITLE
fix: edit account name screen display incorrect account name

### LINKED ISSUE #9829: [Bug]: QR account names are not matching
### Describe the bug

When the user adds QR hardware wallet accounts, the account names should be matching to when the user edits those account names. 

### Expected behavior

QR account name should be matching from the account list dropdown to when editing account name

### Screenshots/Recordings

https://github.com/MetaMask/metamask-mobile/assets/106310394/27427b69-d909-4761-8275-acfc5be39b6f

### Steps to reproduce

1. Recover from SRP
2. Click add hardware wallet
3. Click QR based
4. Add 3 accounts from QR-based wallet
5. Click on QR account 1
6. Click on Edit account name
7. Verify the account name matches on the edit screen to the account list

### Error messages or log output

_No response_

### Version

7.24.0 RC1

### Build type

Beta

### Device

iPhone 13

### Operating system

iOS

### Additional context

_No response_

### Severity

sev-3

### PR DESCRIPTION
**Pull request by dawnseeker8**
## **Description**
Issue #9829 When the user adds QR hardware wallet accounts, the account names should be matching to when the user edits those account names.

After debugging and investigation. find out that in account list screen, account name was retrieved from `internalAccounts.accounts` from `AccountController`, (please refer to `useAccounts.ts` for detail).  but `EditAccountName.tsx` file, the `account name` is from  `identities` from `PreferencesController` and those two values are difference after QR code and ledger imported. after deep investigation. `KeyringController 16` upgrade has removed `updateIdentities` method. and that may cause the PreferenceController`s `identities` state is difference to `internalAccounts` state in AccountController. 

in my fix, i have change `EditAccountName.tsx` to get the account name from `internalAccounts` state just like what `useAccounts.ts` did for account list.

## **Related issues**

Fixes: #9829 

## **Manual testing steps**

1. Recover from SRP
2. Click add hardware wallet
3. Click QR based
4. Add 3 accounts from QR-based wallet
5. Click on QR account 1
6. Click on Edit account name
7. Verify the account name matches on the edit screen to the account list

### **Before**
https://github.com/MetaMask/metamask-mobile/assets/106310394/27427b69-d909-4761-8275-acfc5be39b6f

### **After**

https://github.com/MetaMask/metamask-mobile/assets/7315988/65a903c9-bd83-4f21-8ff0-c9adc987b309

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/EditAccountName/EditAccountName.tsx [hunk 1]
```diff
@@ -38,6 +35,8 @@ import { useTheme } from '../../../util/theme';
 import styleSheet from './EditAccountName.styles';
 import { getDecimalChainId } from '../../../util/networks';
 import { useMetrics } from '../../../components/hooks/useMetrics';
+import { selectInternalAccounts } from '../../../selectors/accountsController';
```
Comment (owencraston) [inline app/components/Views/EditAccountName/EditAccountName.tsx:38]: I love this change. We do not need to use both `selectSelectedAddress` and `selectInternalAccounts`. Instead we can get all of the information we need from just using the `selectSelectedInternalAccount` from the accounts controller state. I have a draft PR with tis exact change. Here is the code.

```tsx
// Third party dependencies
import React, { useCallback, useEffect, useState } from 'react';
import { useNavigation } from '@react-navigation/native';
import { useSelector } from 'react-redux';
import { Platform, SafeAreaView } from 'react-native';

// External dependencies
import Text from '../../../component-library/components/Texts/Text/Text';
import { View } from 'react-native-animatable';
import { TextVariant } from '../../../component-library/components/Texts/Text';
import { strings } from '../../../../locales/i18n';
import TextField from '../../../component-library/components/Form/TextField/TextField';
import { formatAddress, getAddressAccountType } from '../../../util/address';

import Button from '../../../component-library/components/Buttons/Button/Button';
import {
  ButtonSize,
  ButtonVariants,
  ButtonWidthTypes,
} from '../../../component-library/components/Buttons/Button';
import { useStyles } from '../../../component-library/hooks';
import { getEditAccountNameNavBarOptions } from '../../../components/UI/Navbar';
import Engine from '../../../core/Engine';
import generateTestId from '../../../../wdio/utils/generateTestId';
import { MetaMetricsEvents } from '../../../core/Analytics';
import { selectChainId } from '../../../selectors/networkController';
import { selectSelectedInternalAccount } from '../../../selectors/accountsController';
import {
  doENSReverseLookup,
  isDefaultAccountName,
} from '../../../util/ENSUtils';
import { useTheme } from '../../../util/theme';
import { toChecksumHexAddress } from '@metamask/controller-utils';

// Internal dependencies
import styleSheet from './EditAccountName.styles';
import { getDecimalChainId } from '../../../util/networks';
import { useMetrics } from '../../../components/hooks/useMetrics';

const EditAccountName = () => {
  const { colors } = useTheme();
  const { trackEvent } = useMetrics();
  const { styles } = useStyles(styleSheet, {});
  const { setOptions, goBack, navigate } = useNavigation();
  const [accountName, setAccountName] = useState<string>();
  const [ens, setEns] = useState<string>();

  const selectedInternalAccount = useSelector(selectSelectedInternalAccount);

  const selectedChecksummedAddress = selectedInternalAccount?.address
    ? toChecksumHexAddress(selectedInternalAccount.address)
    : undefined;

  const chainId = useSelector(selectChainId);

  const lookupEns = useCallback(async () => {
    if (selectedChecksummedAddress) {
      try {
        const accountEns = await doENSReverseLookup(
          selectedChecksummedAddress,
          chainId,
        );

        setEns(accountEns);
        // eslint-disable-next-line no-empty
      } catch {}
    }
  }, [selectedChecksummedAddress, chainId]);

  useEffect(() => {
    lookupEns();
  }, [lookupEns]);

  const updateNavBar = useCallback(() => {
    setOptions(getEditAccountNameNavBarOptions(goBack, colors));
  }, [setOptions, goBack, colors]);

  useEffect(() => {
    updateNavBar();
  }, [updateNavBar]);

  useEffect(() => {
    const name = selectedInternalAccount?.metadata.name;
    setAccountName(isDefaultAccountName(name) && ens ? ens : name);
  }, [ens, selectedInternalAccount?.metadata.name]);

  const onChangeName = (name: string) => {
    setAccountName(name);
  };

  const saveAccountName = async () => {
    if (
      accountName &&
      accountName.length > 0 &&
      selectedInternalAccount?.address
    ) {
      Engine.setAccountLabel(selectedInternalAccount?.address, accountName);
      navigate('WalletView');

      try {
        const analyticsProperties = async () => {
          const accountType = getAddressAccountType(
            selectedInternalAccount?.address,
          );
          const account_type = accountType === 'QR' ? 'hardware' : accountType;
          return { account_type, chain_id: getDecimalChainId(chainId) };
        };
        const analyticsProps = await analyticsProperties();
        trackEvent(MetaMetricsEvents.ACCOUNT_RENAMED, analyticsProps);
      } catch {
        return {};
      }
    }
  };

  return (
    <SafeAreaView style={styles.screen}>
      <View style={styles.inputsContainer}>
        <View style={styles.inputContainer}>
          <Text variant={TextVariant.BodyLGMedium}>
            {strings('address_book.name')}
          </Text>
          <TextField
            value={accountName}
            onChangeText={onChangeName}
            {...generateTestId(Platform, 'account-name-input')}
          />
        </View>
        <View style={styles.inputContainer}>
          <Text variant={TextVariant.BodyLGMedium}>
            {strings('address_book.address')}
          </Text>
          {selectedInternalAccount?.address ? (
            <TextField
              isDisabled
              placeholder={formatAddress(
                selectedInternalAccount?.address,
                'mid',
              )}
            />
          ) : null}
        </View>
      </View>
      <View style={styles.buttonsContainer}>
        <Button
          variant={ButtonVariants.Secondary}
          size={ButtonSize.Lg}
          width={ButtonWidthTypes.Full}
          label={strings('address_book.cancel')}
          onPress={goBack}
          style={styles.cancelButton}
        />
        <Button
          variant={ButtonVariants.Primary}
          size={ButtonSize.Lg}
          width={ButtonWidthTypes.Full}
          label={strings('address_book.save')}
          onPress={saveAccountName}
          style={
            !accountName?.length
              ? { ...styles.saveButton, ...styles.saveButtonDisabled }
              : styles.saveButton
          }
          disabled={!accountName?.length || accountName?.trim() === ''}
          {...generateTestId(Platform, 'save-button')}
        />
      </View>
    </SafeAreaView>
  );
};

export default EditAccountName;
```

Comment (dawnseeker8) [reply app/components/Views/EditAccountName/EditAccountName.tsx]: have applied your suggestion to the latest `EditAccountName.tsx` file. and it work quite well, thank you very much. :)

File: app/components/Views/EditAccountName/EditAccountName.tsx [hunk 2]
```diff
@@ -74,22 +80,30 @@ const EditAccountName = () => {
...
   const saveAccountName = async () => {
-    if (accountName && accountName.length > 0) {
-      Engine.setAccountLabel(selectedAddress, accountName);
+    if (
+      accountName &&
+      accountName.length > 0 &&
+      selectedInternalAccount?.address
+    ) {
+      Engine.setAccountLabel(selectedInternalAccount?.address, accountName);
+      console.log('Account name updated', accountName);
```
Comment (owencraston) [inline app/components/Views/EditAccountName/EditAccountName.tsx:98]: remove console.log

**Review by owencraston — DISMISSED** [review]
Just remove the console.log and then we can merge.

File: app/components/Views/EditAccountName/EditAccountName.tsx (hunk 2 above)
Comment (dawnseeker8) [reply app/components/Views/EditAccountName/EditAccountName.tsx]: have removed all console .log in the code.

**Review by owencraston — APPROVED** [review]

**Comment by vivek-consensys** [comment]
Tested using iPhone 13 using iOS 17.5.1 on [Bitrise build v7.24.0](https://app.bitrise.io/app/be69d4368ee7e86d/build/fc782601-c757-4c6c-bbd4-f1ab103a7f71/artifact/63250b61dd48d55b/p/31601566974aab131ec42cc34dbd7177) :-


https://github.com/MetaMask/metamask-mobile/assets/106310394/3b73a908-22fe-4800-967f-1584a48d984d

**Pull request merged**